db/data_migrate/20190325121200_fix_us_formatted_dates.rb

```diff
@@ -0,0 +1,72 @@
+require 'progress_bar'
+require 'string_utils'
+
+puts 'Fixing submission entries...'
+
+affected_submission_ids = SubmissionEntry
+                          .validated
+                          .distinct
+                          .select(:submission_id)
+                          .joins(:submission)
+                          .where(
+                            "(LENGTH(data->>'Customer Invoice Date') IN (6,7,8) " \
```

Comment:
I don't understand the length of this field.

If the date includes the separator `/`, wouldn't the minimum length for `dd/mm/yy` be 8?

Reply:
The date is sometimes not zero-prefixed - so possible values could be `1/1/18`  or `1/12/18` or `12/12/18`.

Reply:
Thanks! I also thought some dates could contain a four-digit year, or was that only at the first stage of their being ingested?

Reply:
If they contain a four-digit year, they are already in the correct format, so we skip those.